src/test/java/com/rits/tests/cloning/TestCloner.java

```diff
@@ -790,4 +790,60 @@ public void testConcurrentLinkedQueue() {
 		assertNotSame(list, cloned);
 		assertNotSame(list.peek(), cloned.peek());
 	}
+
+	public void testEnumMapWithNullValue() {
+		EnumMap<TestEnum, String> originalMap = new EnumMap<>(TestEnum.class);
+		originalMap.put(TestEnum.A, null);
+
+		EnumMap<TestEnum, String> clonedMap = cloner.deepClone(originalMap);
+		assertTrue("Cloned LinkedHashSet not instanceof EnumMap", clonedMap instanceof EnumMap);
+		assertEquals("Cloned Map not of expected size", 1, clonedMap.size());
+		assertNull("Expected value is null", clonedMap.get(TestEnum.A));
+		assertTrue("Cloned Map doesn't contain key A", clonedMap.containsKey(TestEnum.A));
+		assertEquals("Cloned EnumMap not equal to original EnumMap", originalMap, clonedMap);
+	}
+
+	public void testEmptyEnumMap() {
+		EnumMap<TestEnum, String> originalMap = new EnumMap<>(TestEnum.class);
+		EnumMap<TestEnum, String> clonedMap = cloner.deepClone(originalMap);
+		assertTrue("Cloned LinkedHashSet not instanceof EnumMap", clonedMap instanceof EnumMap);
+		assertEquals("Cloned Map is not empty", 0, clonedMap.size());
+		assertTrue("Cloned Map is not empty", clonedMap.isEmpty());
+		assertEquals("Cloned EnumMap not equal to original EnumMap", originalMap, clonedMap);
+	}
+	
+
+
+	public void testEnumMapWithNonNullValue() {
+		EnumMap<TestEnum, String> originalMap = new EnumMap<>(TestEnum.class);
+		originalMap.put(TestEnum.A, "Hello");

```

Comment:
One case that is not tested is when the values are actually a non-immutable class. Will those be cloned correctly? (Looking at the EnumMapFastCloner code they are cloned, but maybe a test to prove so would be useful in the future)

Reply:
Agreed, I will add it.

Reply:
Added.